Drv/LinuxGpioDriver/LinuxGpioDriverComponentImpl.cpp

```diff
@@ -105,12 +113,13 @@ namespace Drv {
             return -1;
         }
 
-        // TODO check return value
-        if (out_flag) {
-            (void) write(fd, "out", 4);
-        }
-        else {
-            (void) write(fd, "in", 3);
+        const char *dir = out_flag ? "out" : "in";
+        len = strlen(dir);
```

Comment:
Do we need strnlen here?  Strlen can malfunction if the string is not terminated.

Reply:
In this case it's looking to two constant strings "out" and "in". I though making it dynamic was better than hard coding the string length and having somebody accidentally break it into the future.